fdbserver/TestTLogServer.actor.cpp

```diff
@@ -360,14 +380,22 @@ ACTOR Future<Void> startTestsTLogRecoveryActors(TestTLogOptions params) {
 	wait(buildTLogSet(pTLogTestContextEpochOne));
 
 	PromiseStream<Future<Void>> promises;
+
 	pTLogTestContextEpochOne->ls = ILogSystem::fromServerDBInfo(
 	    pTLogTestContextEpochOne->logID, pTLogTestContextEpochOne->dbInfo, false, promises);
 
 	wait(pTLogTestContextEpochOne->sendPushMessages());
 
+	state uint32_t replicaCount = 2; // SOMEDAY: paramaterize
+	state uint32_t teamSize = pTLogTestContextEpochOne->numLogServers / replicaCount;
+	state uint32_t startLogServer = 0;
+	state uint32_t tagID = 0;
+	state uint32_t logID = startLogServer;
 	if (!pTLogTestContextEpochOne->recover) {
-		for (tLogIdx = 0; tLogIdx < pTLogTestContextEpochOne->numLogServers; tLogIdx++) {
-			wait(pTLogTestContextEpochOne->peekCommitMessages(tLogIdx, 0));
+		for (; tagID < pTLogTestContextEpochOne->numTags; tagID++, startLogServer++) {
+			for (; logID < pTLogTestContextEpochOne->numLogServers; startLogServer += teamSize) {
```

Comment:
The loop here must match the in `sendPushMessage()`. Wonder if you can have a simple map of `(version, tag) -> set of logIDs` and use that map for peeking. The map serves as a ground truth.